bool do_pick
string pick_prompt